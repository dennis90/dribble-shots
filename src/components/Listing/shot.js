import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
