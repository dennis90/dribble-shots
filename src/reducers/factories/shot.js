import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
