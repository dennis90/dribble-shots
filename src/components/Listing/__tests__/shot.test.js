import React from 'react'
import {shallow} from 'enzyme'

import Shot from '../shot-footer'
import {ShotFactory} from '../../../reducers/factories/shot'

describe('Shot Component', () => {
  it('should render shot views count', () => {
    const props = ShotFactory.build({})
    const wrapper = shallow(<Shot shot={props} />)
    expect(wrapper).toMatchSnapshot('Shot div and footer')
  })
})
