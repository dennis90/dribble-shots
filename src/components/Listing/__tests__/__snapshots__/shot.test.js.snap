// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shot div and footer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShotFooter
    shot={
        Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="shot-views"
>
        100
</span>,
      "className": "shot-footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": 100,
        "className": "shot-views",
      },
      "ref": null,
      "rendered": 100,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="shot-views"
>
          100
</span>,
        "className": "shot-footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": 100,
          "className": "shot-views",
        },
        "ref": null,
        "rendered": 100,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
